bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/dto/DataPayload.java

```diff
@@ -176,7 +176,7 @@ public int getLongPowerFailCount() {
      * @return current active voltage
      */
     public int getActiveVoltage() {
-        return activeVoltage;
+        return (int) activeVoltage;
```

Comment:
Should it truncate (like here) or round:
```suggestion
        return (int) Math.round(activeVoltage);
```
?

In the example from the linked issue, it seems that 226.800 will be converted to 226.

Reply:
Yes, that is even better. Changed.

Reply:
Most devices do not report the decimal, as probably no one sees the benefit of 3 digit precisison for a voltage in the 220-250 volt range. I would rate a consistent precision /state pattern accross the binding more important.
So i would like to keep the roudning behavior.

Reply:
Keeping the default state pattern at 0 decimal digits makes sense to me, but why would that mandate rounding the reading? Without rounding the user can play around with the state pattern to get higher precision if ever needed.

Reply:
Changed, don't see the point, but also don't want to spend more time on it. Hope all is fine now.

Reply:
Besides the presentation, there is also persistence to consider. And actually, keeping the full precision can help avoiding excessive state updates. It might sound strange, but having more information, can improve moving average algorithms. See for exampe #12461 and #13393. Back then I didn't have enough knowledge and courage to create a transformation profile for this. It might fit right into [Basic Profiles](https://next.openhab.org/addons/transformations/basicprofiles/), so there's a small project there for the future.

Reply:
Thanks, interesting reads.